core/src/test/java/org/apache/iceberg/TestRowDelta.java

```diff
@@ -629,4 +630,74 @@ public void testFastAppendDoesNotRemoveStaleDeleteFiles() {
         files(FILE_A_DELETES),
         statuses(Status.ADDED));
   }
+
+  @Test
+  public void testValidateDataFilesExistWithConflictDetectionFilter() {
```

Comment:
Should we also add a test where the validation fails? It looks like this one just checks that you can do isolated operations but I think we should do a conflicting test as well.

Reply:
Sure, I'll add one.

Reply:
Added a negative test too.